backend/src/apiserver/server/pipeline_server.go

```diff
@@ -377,14 +369,12 @@ func (s *PipelineServer) GetPipelineByName(ctx context.Context, request *apiv2be
 func (s *PipelineServer) listPipelines(ctx context.Context, namespace string, pageToken string, pageSize int32, sortBy string, opts *list.Options, apiRequestVersion string) ([]*model.Pipeline, []*model.PipelineVersion, int, string, error) {
 	// Fill in the default namespace
 	namespace = s.resourceManager.ReplaceNamespace(namespace)
-	if common.IsMultiUserMode() {
-		resourceAttributes := &authorizationv1.ResourceAttributes{
-			Namespace: namespace,
-			Verb:      common.RbacResourceVerbGet,
-		}
-		if err := s.checkAccess(ctx, resourceAttributes); err != nil {
-			return nil, nil, 0, "", util.Wrapf(err, "Failed to list pipelines due to authorization error. Check if you have read permission to namespace %v", namespace)
-		}
+	resourceAttributes := &authorizationv1.ResourceAttributes{
+		Namespace: namespace,
+		Verb:      common.RbacResourceVerbList,
+	}
```

Comment:
It looks very random that resourceAttributes may have 2 to 4 attributes set and it varies in different places. Don't we need `Resource` and `Name` here and in all other places?

Reply:
The way we check authorization actually became more systemic with this PR. We follow the same pattern everywhere and reuse a lot of code. Depending on what we are trying to do (`Verb`) we may or may not specify `Name` and `Namespace`.

In this particular case, it's because we are trying to list all pipelines in a given namespace. Hence no `Name`. The rest is being added in the `canAccessPipeline`.